kratos/python_scripts/print_info_in_file_process.py

```diff
@@ -0,0 +1,118 @@
+        self.instant_previous_plot = 0.0
+        self.integration_point = settings["integration_point_number"].GetInt()
+
+        open_file_aproach = "a"
+        if settings["erase_previous_info"].GetBool():
+            open_file_aproach = "w"
+
+        self.plot_file = open(self.file_name, open_file_aproach)
+        self.plot_file.write("# In this file we print the " + settings["variable_type"].GetString() + " " + settings["variable_name"].GetString() + " in the ModelPart: " + settings["model_part_name"].GetString() + "\n\n")
+        self.plot_file.write("# TIME\t\t" + settings["variable_name"].GetString() + "\n")
+        self.plot_file.close()
+
+
+    def PrintOutput(self):
+        self.SetPreviousPlotInstant()
+        if self.is_nodal_variable_type: # In nodal values we add them
+            for node in self.model_part.Nodes:
+                counter = 0
+                if counter == 0:
+                    array_values = self.GetValueToPrint(node)
+                if counter > 0:
+                    for value in array_values:
+                        value += self.GetValueToPrint(node)[counter]
+                counter += 1
+            self.plot_file = open(self.file_name, "a")
+            self.plot_file.write("{0:.4e}".format(self.__GetTime()).rjust(11) + "\t")
+            for value in array_values:
+                self.plot_file.write("{0:.4e}".format(value).rjust(11) + "\t")
+            self.plot_file.write("\n")
+            self.plot_file.close()
+        else: # elemental information, not adding values
+            for elem in self.model_part.Elements:
+                counter = 0
+                if counter == 0:
+                    array_values = self.GetValueToPrint(elem)
+                if counter > 0:
+                    for value in array_values:
+                        value += self.GetValueToPrint(elem)[counter]
+                counter += 1
+            self.plot_file = open(self.file_name, "a")
+            self.plot_file.write("{0:.4e}".format(self.__GetTime()).rjust(11) + "\t")
+            for value in array_values:
+                self.plot_file.write("{0:.4e}".format(value).rjust(11) + "\t")
+            self.plot_file.write("\n")
+            self.plot_file.close()
+
+
+    def IsOutputStep(self):
+        if self.output_control_type == "step":
+            return self.model_part.ProcessInfo[KratosMultiphysics.STEP] - self.instant_previous_plot >= self.output_interval
+        else:
+            return self.__GetTime() - self.instant_previous_plot >= self.output_interval
+
+    def SetPreviousPlotInstant(self):
+        if self.output_control_type == "step":
+            self.instant_previous_plot = self.model_part.ProcessInfo[KratosMultiphysics.STEP]
+        else:
+            self.instant_previous_plot = self.__GetTime()
+
+    def GetValueToPrint(self, Entity):
+        if self.is_nodal_variable_type:
+            return Entity.GetSolutionStepValue(self.variable)
```

Comment:
In here you're assuming that the variable to print is historical. Maybe you can modify the `variable_type` field to support `nodal_historical` and `nodal_non_historical` entries.

Reply:
added I think

Reply:
The point on doing the check like this is that if the same variable is in both databases, something that is not common but perfectly valid, you'll be always printing the historical one. I'd directly ask the user for the origin database (as we do in the output processes).

Reply:
ok now I understand, let me solve it